keras/layers/core.py

```diff
@@ -16,21 +16,65 @@
 srng = RandomStreams(seed=np.random.randint(10e6))
 
 class Layer(object):
-    def __init__(self):
+    def __init__(self, name, prev, input_dim):
         self.params = []
+        self.name = name
+        self.prev = prev
+
+        if type(self.prev) is str:
+            self.prev = [self.prev] # single string or a list of strings under one op
+
+        self.input_dim = input_dim
+        self.output_dim = (None, )
+
+    def setup(self):
+        '''
+            Contains code that depends on values that are optional but can be understood from context (ex: input_dim)
+        '''
+        return
+
+    def connect(self, node):
+        self.previous = node
```

Comment:
As a standard, all dimensions are passed around as tuples. It can be in any combination, but the calculations should hold good through the layers.



self.prev is the name of the previous layer(as a string)


self.previous is the actual previous layer object.

On Sat, Jun 27, 2015 at 12:01 AM, Alex Rubinsteyn
<notifications@github.com> wrote:

>> +        self.prev = prev
>> +
>> +        if type(self.prev) is str:
>> +            self.prev = [self.prev] # single string or a list of strings under one op
>> +
>> +        self.input_dim = input_dim
>> +        self.output_dim = (None, )
>> +
>> +    def setup(self):
>> +        '''
>> +            Contains code that depends on values that are optional but can be understood from context (ex: input_dim)
>> +        '''
>> +        return
>> +
>> +    def connect(self, node):
>> +        self.previous = node
> It's a little confusing to have both `self.prev` and `self.previous`, what's the difference between them?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/fchollet/keras/pull/281/files#r33382585